pkg/event/event.go

```diff
@@ -77,16 +77,24 @@ type Recorder interface {
 type APIRecorder struct {
 	kube        record.EventRecorder
 	annotations map[string]string
+	filterFns   []FilterFn
 }
 
+type FilterFn func(obj runtime.Object, e Event) bool
+
 // NewAPIRecorder returns an APIRecorder that records Kubernetes events to an
 // APIServer using the supplied EventRecorder.
-func NewAPIRecorder(r record.EventRecorder) *APIRecorder {
-	return &APIRecorder{kube: r, annotations: map[string]string{}}
+func NewAPIRecorder(r record.EventRecorder, filterFns ...FilterFn) *APIRecorder {
```

Comment:
```suggestion
func NewAPIRecorder(r record.EventRecorder, fns ...FilterFn) *APIRecorder {
```

Nit: I prefer a shorter var name. It's a short function and the `FilterFn` type in the signature lets you know what it's for.

Reply:
agreed, done